### PR TITLE
ICU-22220 BRS73RC Update version number and regenerate configure v73.1

### PR DESCRIPTION
**Pull request by rp9-next**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22220
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

File: icu4j/build.properties [hunk 1]
```diff
@@ -7,6 +7,6 @@
 api.report.version = 73
 api.report.prev.version = 72
 release.file.ver = 73rc
-api.doc.version = 73 Release Candidate
-maven.pom.ver = 73.0.1-SNAPSHOT
+api.doc.version = 73
+maven.pom.ver = 73.1-SNAPSHOT
```
Comment (markusicu) [inline icu4j/build.properties:11]:
```suggestion
maven.pom.ver = 73.1
```

**Review by pedberg-icu — APPROVED** [review]
Looks good, thanks!

**Review by markusicu — APPROVED** [review]
lgtm tnx

@rp9-next did you check if the .jar files actually have meaningful changes (other than differences in zip compression)?

**Comment by rp9-next** [comment]
The last time we checked, while two of the JAR files were exactly same (apart from the "Created By" changes in the MANIFEST.MF)
I believe there were changes with respect to the version number encoded in "*.cnv" and some other changes in ".res" files in testdata.jar

**Pull request merged**